docker/compose/docker-compose_hadoop284_hive233_spark244.yml

```diff
@@ -288,6 +288,8 @@ services:
       - ./hadoop.env
     depends_on:
       - sparkmaster
+    ports:
+      - '4042:4040'
```

Comment:
I had issues since we expose 4040 from adhoc-1. But we run spark shell or spark-submit from adhoc-2. So, I exposed using a diff port no to avoid clash

Reply:
Yeah.  What I mean is you can expose the internal 4042 port to the external 4042 port instead of 4040 which is the current config, to make the port numbers the same.